src/main/java/org/broadinstitute/hellbender/utils/genotyper/ReadLikelihoods.java

```diff
@@ -639,120 +641,28 @@ public boolean addMissingAlleles(final Collection<A> candidateAlleles, final dou
         return result;
     }
 
-
-    /**
-     * Perform marginalization from an allele set to another (smaller one) taking the maximum value
-     * for each read in the original allele subset.
-     *
-     * @param newToOldAlleleMap map where the keys are the new alleles and the value list the original
-     *                          alleles that correspond to the new one.
-     * @return never {@code null}. The result will have the requested set of new alleles (keys in {@code newToOldAlleleMap}, and
-     * the same set of samples and reads as the original.
-     *
-     * @param overlap if not {@code null}, only reads that overlap the location (with unclipping) will be present in
-     *                        the output read-collection.
-     *
-     * @throws IllegalArgumentException is {@code newToOldAlleleMap} is {@code null} or contains {@code null} values,
-     *  or its values contain reference to non-existing alleles in this read-likelihood collection. Also no new allele
-     *  can have zero old alleles mapping nor two new alleles can make reference to the same old allele.
-     */
-    public <B extends Allele> ReadLikelihoods<B> marginalize(final Map<B, List<A>> newToOldAlleleMap, final Locatable overlap) {
```

Comment:
FYI I have decomposed this method in two .... marginalization only and filtering reads (removeReads below) only. Just trying to make it a bit more modular. This change is not necessary for the PR but I though it is a good idea and took the liberty to implement it.

Reply:
Much nicer, thank you!